contributors/design-proposals/storage/dynamic_volume_limit.md

```diff
@@ -0,0 +1,119 @@
+# Dynamic attached volume limits
+
+## Goals
+
+Currently the number of volumes attachable to a node is either hard-coded or only configurable via an environment variable. Also
+existing limits only apply to well known volume types like EBS, GCE and is not available to all volume plugins.
+
+This proposal enables any volume plugin to specify those limits and also allows same volume type to have different volume
+limits depending on type of node.
+
+## Implementation Design
+
+### Prerequisite
+
+* This feature will be protected by an alpha feature gate, so as API and CLI changes needed for it. We are planning to call
+  the feature `DynamicVolumeThreshold`.
+
+### API Changes
+
+There is no API change needed for this feature. However existing `node.Status.Capacity` and `node.Status.Allocatable` will
+be extended to cover volume limits available on the node too.
+
+The key name that will store volume will be start with prefix `storage-limits-`. The volume limit key will respect
+format restrictions applied to Kubernetes Resource names. Volume limit key for existing plugins might look like:
+
+
+* `storage-limits-aws-ebs`
+* `storage-limits-gce-pd`
+* `storage-limits-csi-<driver_name>`
+
+`IsScalarResourceName` check will be extended to cover storage limits:
+
+```go
+func IsStorageResourceName(name v1.ResourceName) bool {
```

Comment:
`StorageResource` is very generic and could mean pretty much anything related to storage, like number of bytes, number of devices, etc. Why not calling it `AttachableVolumes` or `VolumeLimit` or something like that?

Reply:
I agree with being specific (both in the helper method name and in the actual resource prefix in the proposed API)

Reply:
Agreed. Will rename this function and also I think I will rename the actual key being used to `storage-attach-limits-` to make it more clear.

Reply:
Thanks, @gnufied. I think for the key `storage-volume-limits-` is better.

Reply:
fixed.